common/setups/rasr/util/decode.py

```diff
@@ -0,0 +1,43 @@
+__all__ = [
+    "BaseRecognitionParameters",
+    "PriorArgs",
+    "SearchJobArgs",
+    "Lattice2CtmArgs",
+    "ScorerArgs",
+]
+
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, TypedDict
+
+from sisyphus import tk
+
+from ..config.am_config import TransitionTdp, SilenceTdp
+
+
+class BaseRecognitionParameters(TypedDict):
+    am_scales: List[float]
+    lm_scales: List[float]
+    pronunciation_scales: List[float]
+    transition_tdp: List[TransitionTdp]
+    silence_tdp: List[SilenceTdp]
+
+
+@dataclass()
+class PriorArgs:
```

Comment:
why is this a dataclass and the rest a typed dict?

Reply:
For `PriorArgs` I wanted to have the option to add methods. For example to add prior computation "povey" style into the class.
@Marvin84

Reply:
Similarly to the tdps we can also have a flexible prior computation class with following options: 

1.  Average on the train data using the model (povey)
2.  Estimate based on a Viterbi best path, i. e. alignment
3.  Estimate from transcription with specified heuristics based on average phoneme duration

Reply:
Yes, in `PriorArgs` or somewhere else.. didnt think about the design yet.